File: tests/phase7/lib_render_fixtures.js
/* أدوات مشتركة لاختبارات المرحلة 7. */
const fs=require('fs'), _np=require('path');
const HERE=__dirname, ROOT=_np.resolve(HERE,'..','..');
function base(){ return JSON.parse(fs.readFileSync(
  _np.join(ROOT,'tests','phase3','fixtures','base_fixtures.json'),'utf8')); }
function render(){ return JSON.parse(fs.readFileSync(
  _np.join(HERE,'fixtures','render_fixtures.json'),'utf8')); }
function all(){ const o={}, b=base(), r=render();
  Object.keys(b).forEach(k=>{o[k]=b[k];});
  Object.keys(r).forEach(k=>{o[k]=r[k];});
  return o; }
module.exports={base,render,all,ROOT,HERE};

File: tests/phase7/lib_render_fixtures.py
# -*- coding: utf-8 -*-
"""أدوات مشتركة لاختبارات المرحلة 7 في بايثون."""
import json
import os

HERE = os.path.dirname(os.path.abspath(__file__))
ROOT = os.path.dirname(os.path.dirname(HERE))


def base():
    with open(os.path.join(ROOT, 'tests', 'phase3', 'fixtures',
                           'base_fixtures.json'), encoding='utf-8') as f:
        return json.load(f)


def render():
    with open(os.path.join(HERE, 'fixtures', 'render_fixtures.json'),
              encoding='utf-8') as f:
        return json.load(f)


def all_models():
    o = {}
    o.update(base())
    o.update(render())
    return o
